core/src/main/java/bio/terra/pearl/core/dao/workflow/ParticipantTaskDao.java

```diff
@@ -71,6 +72,45 @@ public Optional<ParticipantTask> findTaskForActivity(UUID ppUserId, UUID studyEn
         );
     }
 
+
+    public List<ParticipantTask> findAllTasksForActivity(UUID ppUserId, UUID studyEnvironmentId, String activityStableId) {
+        return jdbi.withHandle(handle ->
+                handle.createQuery("""
+                                select * from %s 
+                                where portal_participant_user_id = :ppUserId
+                                and target_stable_id = :activityStableId 
+                                and study_environment_id = :studyEnvironmentId 
+                                order by created_at desc""".formatted(tableName)
+                        )
+                        .bind("ppUserId", ppUserId)
+                        .bind("activityStableId", activityStableId)
+                        .bind("studyEnvironmentId", studyEnvironmentId)
+                        .mapTo(clazz)
+                        .list()
+        );
+    }
+
+    public Optional<ParticipantTask> findTaskForActivityWithCompletionTime(UUID ppUserId, UUID studyEnvironmentId, String activityStableId, Instant completedAt) {
+        return jdbi.withHandle(handle ->
+                // Note: 43200 seconds is 12 hours. Any task that was completed within 12 hours of the given time is considered a match
+                handle.createQuery("""
+                                select * from %s
+                                where portal_participant_user_id = :ppUserId
+                                and target_stable_id = :activityStableId
+                                and study_environment_id = :studyEnvironmentId
+                                and ABS(EXTRACT(EPOCH FROM completed_at) - EXTRACT(EPOCH FROM :completedAt::timestamp)) < 43200
```

Comment:
can we pull this bit of query string into a helper function?  I suspect we'll want "queryStringForApproximateDateMatch()" in some other places too